### ISSUE TITLE
TinyMCE5.0.3>Dialog>htmlpanel type does not render the style elements such as bold, custom class etc.

### ISSUE THREAD

**Issue by ss2020a** [tinymce/tinymce#6312]
**What is the current behavior? Describe the bug**
when I place html with ordered list or bold etc within htmlpanel of a dialog , dialog is rendered without those styles or ordered list. 

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**

https://fiddle.tiny.cloud/o0haab

In the above example, html: 'This is about <b>cat</b>'   , 

the term cat is not rendered in bold.  Similarly, any custom class, ol etc are removed and rendered as flat/plain text.  I did try with valid_elements, extended_valid_elements configuration but they did not work either.

Anyway, I could get my custom html inside the htmlpanel rendered correctly without influenced by tox classes?

**What is the expected behavior?**

custom html styles, custom html classes inside the htmlpanel content should be honored.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

5.0.3, chrom latest.

**Comment by lostkeys** [comment]
Hi @ss2020a! Yes they are unstyled by the reset styles and we sort of forgot to restore them for core HTML tags like <strong>, <em> and some others. We needed the HTML panel component for the help dialog and only added the styles required for that particular use-case. I'll create a ticket to restore the styles for common basic HTML tags.

**Comment by ss2020a** [comment]
Hi @lostkeys ,  thank you for the reply.  Is there a work-around or a patch I can do temporarily until  a fix is available?  I can even add it as a javascript to postrender or whatever possibilities available.  Appreciate your help on this. thanks

**Comment by lostkeys** [comment]
It's an easy fix, just add your own tox styles like this on the same page as TinyMCE

```
.tox .tox-dialog__body-content .tox-form__group b {
	font-weight: bold;
}

.tox .tox-dialog__body-content .tox-form__group ul {
	list-style: decimal;
}
```

Here is a working demo: https://fiddle.tiny.cloud/t0haab/1

**Comment by ss2020a** [comment]
@lostkeys , thanks a lot. Is there a way, I could put this style part of the css? I tried with css but it did not work whereas it worked when I used style just above the form html similar to how you have demonstrated.

Is there a way, I could tell tinymce not to apply tox classes inheritance to the htmlpanel content or the top most div and its children?

Because, my html has something similar to below and its content could change dynamically as I get it through an ajax call.  some closer sample would be:

        <ol style="list-style-type: upper-alpha;">
            <li>This is line 1</li>
            <li>This is line 2
                <ol style="list-style-type: lower-roman;">
                    <li>This is subline</li>
                    <li>This is subline
                        <ol style="list-style-type: lower-alpha;">
                            <li>This is  subline level 3</li>
                            <li>this is subline level 3</li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ol>



        <table style="border-collapse: collapse;" border="1">
            <tbody>
                <tr>
                    <th style="width: 23.4077%; background-color: #2dc26b; border: 1px groove #3598db;">Sl.No</th>
                    <th style="width: 23.4077%; background-color: #2dc26b; border: 1px groove #3598db;">
                        <p>Firstname</p>
                    </th>
                    <th style="width: 23.4077%; background-color: #2dc26b; border: 1px groove #3598db;">Lastname</th>
                    <th style="width: 23.4077%; background-color: #2dc26b; border: 1px groove #3598db;">Age</th>
                </tr>
                <tr>
                    <td style="width: 23.4077%; text-align: center;">1</td>
                    <td style="width: 23.4077%;">John</td>
                    <td style="width: 23.4077%;">Smith</td>
                    <td style="width: 23.4077%;">22</td>
                </tr>
                <tr>
                    <td style="width: 23.4077%; text-align: center;">2</td>
                    <td style="width: 23.4077%;">Julie</td>
                    <td style="width: 23.4077%;"><span style="background-color: #f1c40f;"><strong>John</strong></span>
                    </td>
                    <td style="width: 23.4077%;">18</td>
                </tr>
                <tr>
                    <td style="width: 23.4077%; text-align: center;">3</td>
                    <td style="width: 23.4077%;">Matt</td>
                    <td style="width: 23.4077%;">Wilson</td>
                    <td style="width: 23.4077%;">2</td>
                </tr>
                <tr>
                    <td style="width: 23.4077%; text-align: center;">4</td>
                    <td style="width: 23.4077%;">John</td>
                    <td style="width: 23.4077%;">Sel</td>
                    <td style="width: 23.4077%;">55</td>
                </tr>
            </tbody>
        </table>


How to apply the native style of the fetched html within the htmlpanel in this case?

Thank you

**Comment by lostkeys** [comment]
1. Yes you can put it in your CSS but it must have higher specificity. An easy fix would be to just use `!important` flags.
2. No, there is no way to have the HTML panel break out of the .tox namespace as that is how CSS works. There is no "stop cascading here" in CSS. Styles will cascade down the tree to the leaf. This is also why we don't support or encourage developers addin their own styles in the TinyMCE UI as changes we do can cascade down into the custom styles and break it.
3. For the list, just apply the styles as described earlier. For the table you should use the `.tox-dialog__table` class. Inspect the help dialog for more details.

That said, you should investigate the [iframe dialog panel component](https://www.tiny.cloud/docs/ui-components/dialogcomponents/#iframe) because the htmlpanel is just for really really basic text-based things like help texts and similar to what we do in the help dialog.

**Comment by ss2020a** [comment]
@lostkeys , thank you for the detailed reply. It helped a lot. I tried with 5.0.3 version in  https://fiddle.tiny.cloud/t0haab/3  and looks good.  I think, I can take it from here. I guess, I can use custom css within the iframe etc without any issues as it is an iframe.

One last question, in case if I choose to use htmlpanel and override the tox styles similar to what you had suggested initially, would not it affect tinymce's OOB features in case the updated styles affects the intended tinymce's OOB functionalities such as dialog header etc?

Many Thanks

**Comment by lostkeys** [comment]
Yes, encapsulating HTML, CSS and JS is what the iframe is good for 😄
I don't know what "OOB" means, but all manual changes you do to TinyMCE complicates the update process and that's why we don't recommend these types of overrides. We want TinyMCE to be easy to upgrade so you don't have to get stuck on a now 1,5 years old version. You are missing out on a lot of bug fixes, improvements and security patches.

**Comment by ss2020a** [comment]
@lostkeys , sure noted. OOB=Out-of-the-box (ie tinymce's original features without any custom stuff). Thanks for all your help. Much appreciated..

**Comment by lostkeys** [comment]
@ss2020a! Ah, I should have figured that out. Yes, depending on what you are doing there might be issues with your styles. If we do a form component that uses `<ol>` to do something, like a rating control, the styles you added would likely break that new control. And changes we do higher up might cascade down to your custom CSS and breaking your customizations. That is the main reason we don't actively give developers the possibility to add custom classes, id:s or other hooks for developers to override the look and behavior of TinyMCE.
